src/nf/nf_activation_3d.f90

```diff
@@ -105,52 +114,79 @@ pure function relu_prime(x) result(res)
     end where
   end function relu_prime
 
-  pure function sigmoid(x) result(res)
+  pure function leaky_relu(x, alpha) result(res)
+    !! Leaky Rectified Linear Unit (Leaky ReLU) activation function.
+    real, intent(in) :: x(:,:,:)
+    real, intent(in) :: alpha
+    real :: res(size(x,1),size(x,2),size(x,3))
+    res = max(alpha*x, x)
+  end function leaky_relu
+
+  pure function leaky_relu_prime(x, alpha) result(res)
+    ! First derivative of the Leaky Rectified Linear Unit (Leaky ReLU) activation function.
+    real, intent(in) :: x(:,:,:)
+    real, intent(in) :: alpha
```

Comment:
It should be `optional` everywhere.
In this case, I would use something like the `stdlib` `optval` function.

Reply:
I'd like to evaluate the performance impact of using `optval` since an activation function is invoked many times and are at the very bottom of the call stack. It would be a good opportunity to use stdlib as a dependency, and via fpm it will be easy. But we'd need to add it as a CMake dependency, which will require a bit more CMake code here. I'm wary of making this PR any larger because it's @Spnetic-5's first PR.

Once we do `activation_params` derived type (in a separate PR), there's a technique which will allow us to not rely on `present` or `optval`.

Reply:
Good points! I agree with you.